rest-openehr/src/main/java/org/ehrbase/rest/BaseController.java

```diff
@@ -113,15 +117,38 @@ public Map<String, Map<String, String>> add2MetaMap(
         return metaMap;
     }
 
+    /**
+     * This method is intended to return the base URI, however it is currently returning the request URI.
+     *
+     * @deprecated This method is deprecated and should not be used in favor of a more updated version.
+     */
+    @Deprecated(forRemoval = true)
     protected String getBaseEnvLinkURL() {
-
         return ServletUriComponentsBuilder.fromCurrentRequest().build().toUriString();
     }
 
-    protected URI getRequestUri() {
-        return ServletUriComponentsBuilder.fromCurrentRequestUri().build().toUri();
+    protected String getContextPath() {
+        return ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString();
     }
 
+    @Value(API_CONTEXT_PATH_WITH_VERSION)
+    protected String apiContextPathWithVersion;
+
+    /**
+     * Returns a URI for list of segments.
+     * The segments are appended to the base path and encoded to ensure safe usage in a URI.
+     * This method uses the UriComponentsBuilder from Spring Framework to construct the URI.
```

Comment:
I suppose mentioning UriComponentsBuilder is not needed

Reply:
done